app/javascript/react/components/ThresholdConfigurator/ThresholdConfigurator.tsx

```diff
@@ -0,0 +1,153 @@
+  initialThresholds: number[];
+}
+
+const ThresholdsConfigurator: React.FC<ThresholdsConfiguratorProps> = ({
+  initialThresholds,
+}) => {
+  const [thresholds, setThresholds] = useState(initialThresholds);
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const [thumbPositions, setThumbPositions] = useState<number[]>([]);
+  const sliderRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (sliderRef.current) {
+      setSliderWidth(sliderRef.current.offsetWidth);
+    }
+    const [min, low, middle, high, max] = thresholds;
+    const lowThumb = calculateThumbPosition(low, min, max, sliderWidth);
+    const middleThumb = calculateThumbPosition(middle, min, max, sliderWidth);
+    const highThumb = calculateThumbPosition(high, min, max, sliderWidth);
+
+    setThumbPositions([lowThumb, middleThumb, highThumb]);
+  }, [thresholds, sliderWidth]);
+
+  const handleInputChange = (index: number, value: string) => {
+    const numericValue = Number(value);
+    const newValues = [...thresholds];
+    newValues[index] = numericValue;
+    setThresholds(newValues);
+  };
+
+  const handleMouseMove =
+    (index: number, startX: number, startValue: number) =>
+    (moveEvent: globalThis.MouseEvent) => {
+      const dx = moveEvent.clientX - startX;
+      const newPercentage =
+        calculateThumbPercentage(startValue, thresholds[0], thresholds[4]) +
+        dx / sliderWidth;
+      const newValue = Math.round(
+        thresholds[0] + newPercentage * (thresholds[4] - thresholds[0])
+      );
+
+      setThresholds((prevThresholds) => {
+        const newThresholds = [...prevThresholds];
+        newThresholds[index + 1] = Math.min(
+          Math.max(newValue, thresholds[0]),
+          thresholds[4]
+        );
+        return newThresholds;
+      });
+    };
+
+  const handleMouseUp =
+    (moveHandler: (moveEvent: globalThis.MouseEvent) => void) => () => {
+      document.removeEventListener("mousemove", moveHandler as EventListener);
+      document.removeEventListener(
+        "mouseup",
+        handleMouseUp(moveHandler) as EventListener
+      );
+    };
+
+  const handleMouseDown =
+    (index: number) => (event: React.MouseEvent<HTMLInputElement>) => {
+      const startX = event.clientX;
+      const startValue = thresholds[index + 1];
+      const moveHandler = handleMouseMove(index, startX, startValue);
+
+      document.addEventListener("mousemove", moveHandler as EventListener);
+      document.addEventListener(
+        "mouseup",
+        handleMouseUp(moveHandler) as EventListener
+      );
+    };
+
+  return (
+    <S.Container ref={sliderRef}>
+      <S.NumberInput
+        type="number"
+        value={thresholds[0]}
+        onChange={(e) => handleInputChange(0, e.target.value)}
+      />
+      {thresholds.slice(1, -1).map((value, index) => (
+        <React.Fragment key={index}>
+          <S.RangeInput
+            $min={thresholds[0]}
+            $max={thresholds[4]}
```

Comment:
Hmm, I think that in iOS we would avoid doing that, we were using a struct just for thresholds. I'm not saying it's the best approach but I remember discussion that keeping them as an array and getting values like that is error prone.  
But tbh I don't know what's the best approach in frontend, feel free to ignore :D